syntax = "proto3";

service Grpc {

  // Server Streaming
	rpc ServerClient (Request) returns (stream Response);

  // Server Streaming
	rpc ClientServer (stream Request) returns (Response);

  // Both Streaming
	rpc InteractingStream (stream Request) returns (stream Response);

}

message Request {
  optional string address = 1;
  optional int32 port = 2;
  optional string request = 3;
}

message Response {
  optional string response = 1;
}
